sklearn/model_selection/_search.py

```diff
@@ -1107,6 +1182,16 @@ class GridSearchCV(BaseSearchCV):
         .. versionchanged:: 0.21
             Default value was changed from ``True`` to ``False``
 
+    use_warm_start : str or list of str, optional
+        The parameters named here will be searched over without clearing the
+        estimator state in between.  This allows efficient searches over
+        parameters where ``warm_start`` can be used. The user should also set
+        the estimator's ``warm_start`` parameter to True.
+
+        Candidate parameter settings will be reordered to maximise use of this
+        efficiency feature.
```

Comment:
I think its worth thinking about how to get this working for `RandomSearchCV`, so all the `*SearchCV` has a consistent API for defining the correct order.

What do you think of something very explicit like the following?

```python
HalvingGridSearchCV(..., use_warm_start={"n_estimators": "increasing"})
```

In the future, if we have this information inside estimator tags, we can have a `use_warm_start='auto'`.